python/metrics/src/hydrotools/metrics/_validation.py

```diff
@@ -100,3 +103,65 @@ def raise_for_inconsistent_shapes(
                 array_shape_1=x.shape,
                 array_shape_2=y.shape
                 )
+
+def validate_boolean_categorical_series(
+    series: pd.Series
+    ) -> None:
+    """
+    Validate that series is a boolean categorical variable.
+
+    Parameters
+    ----------
+    series: pandas.Series, required
+        Series to validate.
+
+    Warnings
+    --------
+    UserWarning:
+        Warns if series is not a boolean Categorical and attempts to convert.
+
+    Raises
+    ------
+    pandas.errors.UnsupportedFunctionCall:
+        Raises if series is not a pandas.Series or if series does not contain two categories: True and False
+
+    Returns
+    -------
+    Validated boolean categorical series.
+    """
+    # Check for Series
+    if not isinstance(series, pd.Series):
+        raise pd.errors.UnsupportedFunctionCall(f"{series} is not a pandas.Series")
+
+    # Check for categorical
+    if not hasattr(series, "cat"):
+        message = f"{series} is not a categorical pandas.Series, attempting to convert"
+        warnings.warn(message=message, category=UserWarning)
+        series = series.astype("category")
+
+    # Check for True category
+    no_true_category = True
+    for c in series.cat.categories:
+        if c is True:
+            no_true_category = False
+    if no_true_category:
+        message = f"{series} has no True category, attempting to add"
```

Comment:
what does this print? I would assume that it would print the, series and then `has no True category, attempting to add`. Is that right? Or am I missing something?

Reply:
Correct, though this has been removed.